Lib/zipfile.py

```diff
@@ -2161,6 +2143,18 @@ def _ancestry(path):
         path, tail = posixpath.split(path)
 
 
+_dedupe = OrderedDict.fromkeys
```

Comment:
dict is ordered and using `dict.fromkeys` could be more efficient while deduping long lists.

```
➜  cpython git:(master) ✗ ./python -m timeit -s 'from collections import OrderedDict; data = range(10000)' 'list(OrderedDict.fromkeys(data))'
100 loops, best of 5: 3.25 msec per loop
➜  cpython git:(master) ✗ ./python -m timeit -s 'data = range(10000)' 'list(dict.fromkeys(data))'                                            
200 loops, best of 5: 1.64 msec per loop
```

Reply:
Oh, good tip. I'd forgotten about that. I'll update the backport implementation with that first.

Reply:
Implemented in jaraco/zipp@19599b7.

Reply:
I could have presented a slightly incorrect benchmark since I guess `dict.fromkeys` seems efficient in a loop where it reuses the keys but for one time use I guess it's of same speed with OrderedDict. Nevertheless, I consider this to be clean and also the size of the dict object is lower than OrderedDict resulting in memory saving for large dict.